book/continuous_integration.md

```diff
@@ -0,0 +1,192 @@
+
+This workflow will report whether our test code ran successfully for each of the specified Python versions.
+
+### Configure GitHub actions to build and deploy documentation
+
+This book uses the following GitHub Actions configuration to build and deploy the HTML content:
+
+```yaml
+name: Build and deploy book
+
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+
+    - name: Install dependencies
+      run: |
+        pip install -r requirements.txt
+
+    - name: Build the book
+      run: |
+        jb build book -W -v --keep-going && touch ./book/_build/html/.nojekyll
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+    if: startsWith( github.ref, 'refs/tags/v')
+    steps:
+    - name: "Deploy book to GitHub Pages"
+      uses: peaceiris/actions-gh-pages@v3.6.1
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./book/_build/html
+```
+
+This workflow runs whenever a pull request is create or changes are pushed directly to the main branch. It has two jobs - one that builds the book's HTML content and another that deploys the content to this website.
+
+As with the previous example, we start the workflow by setting up an environment with Python. We install the dependencies for the project, which includes `jupyter-book` to build to the book.
+
+Our workflow then builds the book's HTML content, where the workflow will fail if warnings or errors are raised.
+
+In the second job, the book (including the new changes) is deployed to the site that you are reading now. This job needs the build job to have completed successfully before it will run. It will only run if a new Git tag has been created, to indicate a new version of the book. This allows us to accumulate changes on the main branch, before releasing a collection of changes in the next version. This deployment step requires authentication, which is managed by a secret/token that is accessed from the Action's environment.
+
+You might use a similar approach to this to deploy your code's HTML documentation.
+
+### Comprehensive example
+
+You can see a detailed example of CI in practice in the `jupyter-book` project. A recent version of the [`jupyter-book` CI workflow](https://github.com/executablebooks/jupyter-book/blob/6fb0cbe4abb5bc29e9081afbe24f71d864b40475/.github/workflows/tests.yml) includes:
+
```

Comment:
I worry we are catering to GitHub only, would a GitLab example make sense? Or maybe a link to a GitLab equivalent example.

Reply:
yes, I thought this too. I'm not sure if this was a conscious decision in the original. @foster999 is there any reason not to include a GitLab example?

Reply:
resolving this. please see other comment above about examples